src/Installer/dotnetup/WindowsPathHelper.cs

```diff
@@ -0,0 +1,461 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+using System.Security.Principal;
+using Microsoft.Win32;
+
+namespace Microsoft.DotNet.Tools.Bootstrapper;
+
+/// <summary>
+/// Helper class for Windows-specific PATH management operations.
+/// </summary>
+[SupportedOSPlatform("windows")]
+internal sealed class WindowsPathHelper : IDisposable
+{
+    private const string RegistryEnvironmentPath = @"SYSTEM\CurrentControlSet\Control\Session Manager\Environment";
+    private const string PathVariableName = "Path";
+    private const int HWND_BROADCAST = 0xffff;
+    private const int WM_SETTINGCHANGE = 0x001A;
+    private const int SMTO_ABORTIFHUNG = 0x0002;
+
+    private readonly StreamWriter? _logWriter;
+    private readonly string? _logFilePath;
+    private bool _disposed;
+
+    [DllImport("user32.dll", CharSet = CharSet.Auto, SetLastError = true)]
+    private static extern IntPtr SendMessageTimeout(
+        IntPtr hWnd,
+        int Msg,
+        IntPtr wParam,
+        string lParam,
+        int fuFlags,
+        int uTimeout,
+        out IntPtr lpdwResult);
+
+    /// <summary>
+    /// Creates a new instance of WindowsPathHelper with logging enabled.
+    /// </summary>
+    public WindowsPathHelper()
+    {
+        try
+        {
+            string tempPath = Path.GetTempPath();
+            string logFileName = $"dotnetup_path_changes_{DateTime.Now:yyyyMMdd}.log";
+            _logFilePath = Path.Combine(tempPath, logFileName);
+            _logWriter = new StreamWriter(_logFilePath, append: true);
+            _logWriter.AutoFlush = true;
+            LogMessage($"=== WindowsPathHelper session started at {DateTime.Now:yyyy-MM-dd HH:mm:ss} ===");
+        }
+        catch (Exception ex)
```

Comment:
Don't swallow this exception here.  We can rethrow it to add a better message as long as we preserve the inner exception.